hledger/Hledger/Cli/CliOptions.hs

```diff
@@ -454,8 +454,9 @@ replaceNumericFlags = map replace
 -- Also records the terminal width, if supported.
 rawOptsToCliOpts :: RawOpts -> IO CliOpts
 rawOptsToCliOpts rawopts = do
-  let iopts = rawOptsToInputOpts rawopts
-  rspec <- rawOptsToReportSpec rawopts
+  d <- getCurrentDay
+  let iopts = rawOptsToInputOpts d rawopts
+  rspec <- either fail return $ rawOptsToReportSpec d rawopts
```

Comment:
fail in IO is similar to `error` I guess, but what's the error message ? We might need to ensure it's a good one ?

Reply:
The error message is the string from the `Either String`